File: nptelDaad/w2/binary_search.cpp
#include <iostream>

using namespace std;

int binarySearch(int arr[], int len, int x);

int main() {
  int arr[] = {2, 8, 12, 17, 23, 28, 34, 39, 42, 44};
  int len = 10;

  cout << binarySearch(arr, len, 44);

  return 0;
}

int binarySearch(int arr[], int len, int x) {
  int high = len;
  int low = 0;
  int mid;

  while(low < high) {
    mid = (low + high) / 2;

    if(x > arr[mid]) low = mid;
    else if(x < arr[mid]) high = mid;
    else return mid;
  }
  return -1;
}
  

File: bca-sem4/program/python/BinarySearch.py
def binarySearch(list1, n):
    low = 0
    high = len(list1) - 1
    mid = 0

    while low <= high:
        mid = (low + high) // 2

        if list1[mid] < n:
            low = mid + 1
        elif list1[mid] > n:
            high = mid - 1
        else:
            return mid

    return -1

list1 = list(map(int, input("Enter the elements in the list : ").split()))
num = int(input("Enter the element to be searched : "))

res = binarySearch(list1, num)

if res != -1:
    print("The Index of the given number is : " ,res)
else:
    print("The given element not found")
